### DISCUSSION TITLE
Windows BOOL

### DISCUSSION THREAD

**Discussion by joschua-fink** [scala-native/scala-native#3980] [Q&A]
What is the correct way to map a [Windows BOOL](https://learn.microsoft.com/en-us/windows/win32/winprog/windows-data-types) to Scala?

As example `ConvertStringSidToSidW` in [SddlApi.scala](https://github.com/scala-native/scala-native/blob/main/windowslib/src/main/scala/scala/scalanative/windows/SddlApi.scala) in this project returns a Scala Boolean. The [Windows API of the function](https://learn.microsoft.com/en-us/windows/win32/api/sddl/nf-sddl-convertstringsidtosidw) returns a Windows BOOL, which is - as I understand it - defined as an CInt.

When I was playing around with a similiar case both using CInt and Boolean seemed to compile fine and behave the same way. But I am not sure if that is true under all circumstances, because I don't have much experience in using the Windows API and C.

If Boolean has the size of 1 byte and CInt the size of 4 bytes, at least theoretically a larger CInt number (e.g. 256) which could be interpreted as "true" (!= 0), if converted to a single byte, could become 0, which would be interpreted as "false".

**Comment by philwalk** [comment]
In `C` booleans are type `int`, with TRUE and FALSE defined as `1` and `0` respectively.

**Comment by ekrich** [comment]
It looks like on the JVM a boolean is a byte with 0, 1 as values. So maybe that is why using boolean is so seamless.
